sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -670,6 +670,8 @@ case class Window(
   override def output: Seq[Attribute] =
     child.output ++ windowExpressions.map(_.toAttribute)
 
+  override def producedAttributes: AttributeSet = windowOutputSet
```

Comment:
BTW, isn't this a bug of `Window` class implementation itself, @viirya ?

Reply:
I'm wondering if we can have this line for all the other applicable branches.

Reply:
Yea, I think so, although I changed it here for nested column pruning.

Currently it seems not causing other problem, although it is wrong. Looks good to have it in other branches (3.0/2.4) too.